sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -31,10 +30,16 @@ import org.apache.spark.util.ListenerBus
  *
  * Implementations should throw [[NoSuchDatabaseException]] when databases don't exist.
  */
-abstract class ExternalCatalog
-  extends ListenerBus[ExternalCatalogEventListener, ExternalCatalogEvent] {
+trait ExternalCatalog {
```

Comment:
Based on your JIRA comment, can we put `@DeveloperApi` or `@InterfaceStability.Unstable` in this PR?

Reply:
This trait is still an internal one. We have not make it public yet.

Reply:
I see. Thank you for confirming, @gatorsmile .

Reply:
Hopefully, the next release.